src/main/java/gregtech/common/covers/filter/SimpleItemFilter.java

```diff
@@ -134,4 +181,62 @@ private static boolean areItemsEqual(boolean ignoreDamage, boolean ignoreNBTData
         }
         return ignoreNBTData || ItemStack.areItemStackTagsEqual(filterStack, itemStack);
     }
+
+    protected class SimpleItemFilterReader extends BaseItemFilterReader {
+
+        public static final String IGNORE_NBT = "ignore_nbt";
+        public static final String IGNORE_DAMAGE = "ignore_damage";
+
+        public SimpleItemFilterReader(ItemStack container, int slots) {
+            super(container, slots);
+        }
+
+        protected void setIgnoreDamage(boolean ignoreDamage) {
+            getStackTag().setBoolean(IGNORE_DAMAGE, ignoreDamage);
+            markDirty();
+        }
+
+        protected void setIgnoreNBT(boolean ignoreNBT) {
+            getStackTag().setBoolean(IGNORE_NBT, ignoreNBT);
+            markDirty();
+        }
+
+        public boolean isIgnoreDamage() {
+            if (!getStackTag().hasKey(IGNORE_DAMAGE))
+                setIgnoreDamage(true);
```

Comment:
Maybe `IGNORE_DAMAGE` (and `IGNORE_NBT`) should be inverted here: change them to `RESPECT_X` and have the default include no nbt tag (so a default false value) representing respecting the attribute, therefore by default still ignoring it. That way, we don't have to add nbt tags to represent the default state.